File: 1457-pseudo-palindromic-paths-in-a-binary-tree/1457-pseudo-palindromic-paths-in-a-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    
    int cnt=0;
    void dfs(TreeNode* root,vector<int>ans){
        
        
            if(!root->left and !root->right){
                   ans[root->val]+=1;
                    int flag=0;
                    for(int i=0;i<ans.size();i++){
                            if((ans[i]%2!=0 and flag==0) or (ans[i]%2==0)){
                                        if(ans[i]%2!=0){
                                                flag=1;
                                        }
                                        
                            }
                            else{
                                    
                                        return;
                                
                            }
                        
                    }
                    cnt+=1;
            }
        
            ans[root->val]+=1;
            if(root->left) dfs(root->left,ans);
            if(root->right) dfs(root->right,ans);
            
            
        
    }
    
    
    int pseudoPalindromicPaths (TreeNode* root) {
        
            vector<int>ans(10,0);
            dfs(root,ans);
            return cnt;
        
        
    }
};

File: 1457-pseudo-palindromic-paths-in-a-binary-tree/1457-pseudo-palindromic-paths-in-a-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
   
    def pseudoPalindromicPaths (self, root: Optional[TreeNode]) -> int:
        
        
                arr=[0]*9
                print(arr)
                cnt=0
                def palindrome(root,arr):
                    
                    nonlocal cnt
                    if root.left==None and root.right==None:
                            arr[root.val-1]+=1
                            charCnt=sum(arr)
                            if charCnt%2==0:
                                    for i in range(9):
                                            if arr[i]%2!=0:
                                                    return
                                    cnt+=1
                            else:
                                    onlyOne=False
                                    for i in range(9):
                                            if arr[i]%2!=0 and onlyOne==True:
                                                    return
                                            elif arr[i]%2!=0 and onlyOne==False:
                                                    onlyOne=True
                                            
                                    cnt+=1
                            return
                
                    arr[root.val-1]+=1
                    if root.left:
                        palindrome(root.left,arr.copy())
                    if root.right:
                        palindrome(root.right,arr.copy())
                    
              
                palindrome(root,arr)
                return cnt        
                